Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -8,18 +8,64 @@ Android assigns a distinct system identity (Linux user ID and group ID) to every
+    - The `ACCESS_FINE_LOCATION` or `ACCESS_COARSE_LOCATION` permission.
+    - The `ACCESS_WIFI_STATE` permission.
+    - Location services are enabled (under Settings -> Location).
+
+Apps targeting Android 9, i.e. [only API level 28 or higher](https://developer.android.com/about/versions/pie/android-9.0-changes-28 "Behavior changes: apps targeting API level 28+") are affected by the following:
+
+- **Restricted foreground services**: apps using foreground services must request the `FOREGROUND_SERVICE` permission in their Android manifest (it's a normal permission and therefore automatically granted at installation time).
+- **Build serial number deprecation**: the `READ_PHONE_STATE` permission is required now in order to read the device's hardware serial number via [`Build.getSerial`](https://developer.android.com/reference/android/os/Build.html#getSerial() "getSerial") ([`Build.SERIAL`](https://developer.android.com/reference/android/os/Build.html#SERIAL "Build.SERIAL") is now set to "UNKNOWN").
+
+##### Android 10 Changes (Beta)
+
+Android 10 Beta introduces several [user privacy enhancements](https://developer.android.com/preview/privacy/permissions "Android Q privacy: Changes to permissions"). The changes regarding permissions affect to all apps running on Android 10, including those targeting lower API levels.
+
+- **Restricted access to screen contents**: `READ_FRAME_BUFFER`, `CAPTURE_VIDEO_OUTPUT`, and `CAPTURE_SECURE_VIDEO_OUTPUT` permissions are now signature-access only, which prevents silent access to the device's screen contents.
+- **User-facing permission check on legacy apps**: when running an app targeting Android 5.1 (API level 22) or lower for the first time, users will be prompted with a permissions screen where they can revoke access to specific _legacy permissions_ (which previously would be automatically granted at installation time).
+- **Physical activity recognition**: there is a new `ACTIVITY_RECOGNITION` runtime permission for apps that need to detect the user's step count or classify the user's physical activity
+- **Permission groups removed from UI**: despite its misleading name, this change means that apps are not able to [programmatically look up](https://stackoverflow.com/a/35000855 "Link between Android Permissions and Permission Groups") how permissions are grouped (e.g. by using PackageManager's `getAllPermissionGroups` and `queryPermissionsByGroup` methods). Developers may look up manually in the corresponding [platform manifest](https://github.com/aosp-mirror/platform_frameworks_base/blob/android-q-preview-5/core/res/AndroidManifest.xml "Android Q Preview 5 - Platform Manifest") and request them.
```

Comment:
Even though this is great information and thanks for mentioning all of it, we should always keep in mind who our audience is, which is developers (that want to know about security) and penetration testers.
This information above for Android 9 and 10 is about permissions in general and the changes are not directly relevant to our audience. So I would suggest we just keep a link to the Android documentation like this instead:

For all changes related to permissions, please check the documentation for [Android 9 \(API Level 28\)](https://developer.android.com/about/versions/pie/android-9.0-changes-all) and Android [Android 10](https://developer.android.com/preview/privacy/permissions).

Reply:
the same question raises here: should we leave all changes to permissions that directly affect access to sensitive data and/or user privacy?

For the moment, I've removed a couple of implementation related details and left:
- Android 9.0: access to call log, phone number; SSID and BSSID values, device hardware serial number
- Android 10: access to device screen contents, permissions can now be revoked from legacy apps that would be otherwise automatically granted.

Another option would be to summarize these points even more, apply them to the relevant testing sections or if not, directly use your suggestion. Please let me know your thoughts :)

Reply:
It's definitely a great overview and thanks for trimming it down a bit. 
I am just thinking about the future, what about other upcoming Android versions (which of course will also have a lot of permission changes)? If we follow this structure this section will become rather complicated to read with all the different Android versions. 
So we should just keep in mind that either we get rid of some of the older information then in the future (as Jeroen was writing below) or we summarize it then more and give more focus on the recent changes and maybe link to the older changes instead as I was suggesting earlier.
This is also something we need to keep in mind in general for the guide. If we keep adding things the project will become hard to maintain and will have less value to the reader. So this will also be one of our biggest challenges in the future, especially for Android.

Reply:
I agree, we should add a guideline for this, e.g. we define a range of API levels that we cover and on each new major Android release we rotate. It should be a bit easier if we keep up been consistent with the format (using the current approach of "Android X (API level xx)", as this makes it much easier to track across the guide and therefore to update/remove.

The question would also be what to do with the "old stuff", as it might also be a pity that it simply goes away. Maybe we could have something like an archive? we could maintain it in the GitHub repo but outside `Document/` (e.g. under `Archive/`?). In the guide itself we could refer to the archive page in GitHub. e.g. in 0x5h "Testing App Permissions" we would have something like:

"Please refer to the [MSTG archive](https://github.com/OWASP/owasp-mstg/blob/master/Archive/0x05h-Testing-Platform-Interaction.md) for information regarding permission changes in older Android versions."

We can keep the discussion on Slack ;)

Reply:
I rather refer to older tags in git there ;-)